File: prime/src/main/java/org/ostelco/prime/storage/entities/RecordOfPurchaseImpl.java
package org.ostelco.prime.storage.entities;

import lombok.Data;
import lombok.Getter;

import static com.google.api.client.util.Preconditions.checkNotNull;

@Data
public final class RecordOfPurchaseImpl implements RecordOfPurchase {

    @Getter private  final String msisdn;

    @Getter private  final String sku;

    @Getter private  final long millisSinceEpoch;

    public RecordOfPurchaseImpl(
            final String msisdn,
            final String sku,
            final long millisSinceEpoch) {
        this.msisdn = checkNotNull(msisdn);
        this.sku = checkNotNull(sku);
        this.millisSinceEpoch = millisSinceEpoch;
    }
}

File: prime/src/main/kotlin/org/ostelco/prime/storage/entities/RecordOfPurchaseImpl.kt
package org.ostelco.prime.storage.entities


data class RecordOfPurchaseImpl(
        val _msisdn: String,
        val _sku: String,
        val _millisSinceEpoch: Long) : RecordOfPurchase {

    override val msisdn: String
        get() = _msisdn
    override val millisSinceEpoch: Long
        get() = _millisSinceEpoch
    override val sku: String
        get() = _sku
}
